src/Riok.Mapperly/Descriptors/UserMethodMappingExtractor.cs

```diff
@@ -67,7 +67,7 @@ bool isStatic
             BuildParameters(ctx, method, out var parameters)
             && !method.ReturnsVoid
             && (allowPartial || !method.IsPartialDefinition)
-            && isStatic == method.IsStatic;
+            && !(isStatic && !method.IsStatic);
```

Comment:
IMO a `||` condition would be easier to read here: `(!isStatic || method.IsStatic)`